scripts/locale-checks.ts

```diff
@@ -0,0 +1,318 @@
+      console[type](list(type, keys));
+    });
+  }
+};
+
+const getNamespace = (key: string): string => key.split(':')[0];
+const removeNamespace = (key: string): string => key.split(':')[1];
+
+const isObject = (obj: unknown): obj is Record<string, unknown> =>
+  typeof obj === 'object' && obj !== null && !Array.isArray(obj);
+
+const getNestedPropertyNames = (obj: Record<string, unknown>, parentKey?: string): string[] => {
+  const result: string[] = [];
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
+      const currentKey = parentKey ? `${parentKey}.${key}` : key;
+      const currentValue = obj[key];
+      if (isObject(currentValue)) {
+        result.push(...getNestedPropertyNames(currentValue, currentKey));
+      } else {
+        result.push(currentKey);
+      }
+    }
+  }
+  return result;
+};
+
+const findMissingInArray = (arr1: string[] | Set<string>, arr2: string[] | Set<string>): string[] | false => {
+  const missingItem = Array.from(arr1).filter((item) => !Array.from(arr2).includes(item));
+  return missingItem.length ? missingItem : false;
+};
+
+const findMissingInArrayWithNamespacedLocaleKeys = (
+  computedNamespacedKeys: string[],
+  locale: Locale,
+  option?: { reverse?: boolean; localeKeyNamesOnly?: boolean },
+): StringArrayRecord | false => {
+  const groupByNamespace = computedNamespacedKeys.reduce<StringSetRecord>((acc, item) => {
+    const namespace = getNamespace(item);
+    if (!acc[namespace]) acc[namespace] = new Set();
+    acc[namespace].add(removeNamespace(item));
+    return acc;
+  }, {});
+
+  return Object.entries(locale)
+    .filter(([key]) => key != 'id')
+    .filter(([key]) => !option?.localeKeyNamesOnly || (option?.localeKeyNamesOnly && key === 'names'))
+    .reduce<StringArrayRecord | false>((acc, [namespace, obj]) => {
+      const props = getNestedPropertyNames(obj);
+      const missingItem = option?.reverse
+        ? props.filter((item) => !Array.from(groupByNamespace[namespace] ?? []).includes(item))
+        : Array.from(groupByNamespace[namespace] ?? []).filter((item) => !props.includes(item));
+      if (missingItem.length) {
+        if (acc === false) acc = {};
+        acc[namespace] = missingItem;
+      }
+      return acc;
+    }, false);
+};
+
+const findMissingLocalizedItems = (
+  localeKey: string,
+  localeComputedKeys: string[] | Set<string>,
+  option?: { reverse?: boolean; localeKeyNamesOnly?: boolean },
+): StringArrayRecord | false => {
+  // For a region locale (XxXx), merge it with the base locale (Xx) if exists
+  const locale =
+    localeKey.length === 4 && Object.prototype.hasOwnProperty.call(locales, localeKey.slice(0, 2))
+      ? merge(locales[localeKey.slice(0, 2)], locales[localeKey])
+      : locales[localeKey];
+
+  return findMissingInArrayWithNamespacedLocaleKeys(Array.from(localeComputedKeys), locale, option);
+};
+
+const devLocale = { id: 'dev' };
```

Comment:
I didn’t know we can use `dev` as an ID. Where is that defined? I thought it should of the `Locale` type which is defined as `string`, however, it needs to be found in one of the l10n files.

Reply:
At this stage, you can pass whatever you want. Also this script is accessing internal Romcal components, which should not be accessed directly by end users utilizing Romcal. Here for some checks, I don't really need a locale data, so I'm creating a mock object when a locale is required.